src/components/inline_edit/inline_edit.tsx

```diff
@@ -0,0 +1,221 @@
+    } else {
+      setEditViewValue(readViewValue);
+      setIsInEdit(!isInEdit);
+    }
+  };
+
+  const defaultSaveButtonAriaLabel = useEuiI18n(
+    'euiInlineEdit.saveButtonAriaLabel',
+    'Save edit'
+  );
+
+  const defaultCancelButtonAriaLabel = useEuiI18n(
+    'euiInlineEdit.cancelButtonAriaLabel',
+    'Cancel edit'
+  );
+
+  let iconSize: EuiButtonIconProps['iconSize'];
+  let compressed; // affects FieldText compressed field and editView button size
+  switch (size) {
+    case 'xxxs':
+    case 'xxs':
+    case 'xs':
+    case 's':
+      iconSize = 's';
+      compressed = true;
+      break;
+    default:
+      iconSize = 'm';
+      compressed = false;
+      break;
+  }
+
+  const textEditViewElement = (
+    <EuiForm fullWidth>
+      <EuiFlexGroup gutterSize="s">
+        <EuiFlexItem>
+          <EuiFieldText
+            id={inlineEditInputId}
+            value={editViewValue}
+            onChange={editTextViewOnChange}
+            aria-label={inputAriaLabel}
+            autoFocus
+            compressed={compressed}
+            {...(rest as any)}
+          />
+        </EuiFlexItem>
+
+        <EuiFlexItem grow={false}>
+          <EuiFormRow>
+            <EuiButton
+              iconType="check"
+              aria-label={saveButtonAriaLabel || defaultSaveButtonAriaLabel}
+              onClick={saveTextEditValue}
+              color="primary"
+              fill
+              size={compressed ? 's' : 'm'}
+            >
+              Save
+            </EuiButton>
+          </EuiFormRow>
+        </EuiFlexItem>
+
+        <EuiFlexItem grow={false}>
+          <EuiFormRow>
+            <EuiButton
+              iconType="cross"
+              aria-label={cancelButtonAriaLabel || defaultCancelButtonAriaLabel}
+              onClick={() => {
+                setEditViewValue(readViewValue);
+                setIsInEdit(!isInEdit);
+              }}
+              color="primary"
+              size={compressed ? 's' : 'm'}
+            >
+              Cancel
+            </EuiButton>
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiForm>
+  );
+
+  const H: Heading = heading;
+  const textReadViewElement = (
+    <EuiButtonEmpty
+      color="text"
+      iconType="pencil"
+      iconSide="right"
+      autoFocus
+      flush="both"
+      iconSize={iconSize}
+      size={compressed ? 's' : 'm'}
+      onClick={() => {
+        setIsInEdit(!isInEdit);
+      }}
+    >
+      {display === 'title' ? (
+        <EuiTitle size={size as EuiTitleSize}>
+          <H>{readViewValue}</H>
+        </EuiTitle>
+      ) : (
+        <EuiText size={size as EuiTextProps['size']}>{readViewValue}</EuiText>
+      )}
+    </EuiButtonEmpty>
+  );
+
+  /* Current Form Control in View */
+  const currentFormControlInView = isInEdit
+    ? textEditViewElement
+    : textReadViewElement;
+
+  return (
+    <div className={classes} {...rest}>
+      {currentFormControlInView}
+    </div>
+  );
```

Comment:
TBH, I don't know that defining this extra var is really adding that much readability as opposed to just using it in a ternary in JSX:

```suggestion
  return (
    <div className={classes} {...rest}>
      {isInEdit ? textEditViewElement : textReadViewElement}
    </div>
  );
```